### PR TITLE
Adds option to search for RigidBodyAPI on all matching prim paths

### PR DESCRIPTION
**Pull request by kellyguo11**
# Description

When defining `RigidObject` instances with regex that encapsulates heterogeneous objects, the objects may not always have the RigidBodyAPI schema attached to the same prim path in the object hierarchy. Currently, we make this assumption such that we only search for the location of the RigidBodyAPI schema on the first matching prim path of the regex. However, when the assumption doesn't hold on objects that have different topologies, the `RigidObject` instance is not able to find all objects in the scene that match the regex.

This change introduces a new option in the `RigidObjectCfg` that allows for parsing of all matching prim paths instead of just the first matching path. This will incur performance costs when parsing large stages, so it is recommended to only use this option when necessary, when it is desired to have a `RigidObject` that encapsulates a collection of varying objects where the `RigidBodyAPI` schema is attached to different locations.

## Type of change

- New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have run the [`pre-commit` checks](https://pre-commit.com/) with `./isaaclab.sh --format`
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have updated the changelog and the corresponding version in the extension's `config/extension.toml` file
- [ ] I have added my name to the `CONTRIBUTORS.md` or my name already exists there

### CODE REVIEW

**Comment by Mayankm96** [comment]
Doesn't this affect the shape of the tensors? We always expect one rigid body per environment so all the tensors have shape (num_envs, ...). Now if you are allowing multiple prims to be captured, wouldn't the shape become (num_envs * num_rigid_body_per_env, ...)? This seems rather not the right way in that case.

File: source/extensions/omni.isaac.lab/test/assets/test_rigid_object.py [hunk 1]
```diff
@@ -118,6 +168,44 @@ def test_initialization(self):
...
+            with self.subTest(num_cubes=num_cubes, device=device):
+                with build_simulation_context(device=device, auto_add_lighting=True) as sim:
+                    # Generate cubes configs
+                    cube_object_cfg, spawn_cfgs, _ = generate_heterogeneous_cubes_scene(
+                        num_cubes=num_cubes, device=device
+                    )
+
+                    # Spawn objects to scene
+                    for cfg in spawn_cfgs:
+                        RigidObject(cfg)
```
Comment (Mayankm96) [inline source/extensions/omni.isaac.lab/test/assets/test_rigid_object.py]: Why is there a spawn again if the function above generates the heterogenous scene?

**Comment by kellyguo11** [comment]
> Doesn't this affect the shape of the tensors? We always expect one rigid body per environment so all the tensors have shape (num_envs, ...). Now if you are allowing multiple prims to be captured, wouldn't the shape become (num_envs * num_rigid_body_per_env, ...)? This seems rather not the right way in that case.

hmm ya this could happen if the regex encapsulates multiple objects in an environment. If we want to keep it to a single body per env, I guess we can use the same logic to get the first template path and traverse each tree from there to find the RigidBodyAPI.

File: source/extensions/omni.isaac.lab/test/assets/test_rigid_object.py (hunk 1 above)
Comment (kellyguo11) [reply source/extensions/omni.isaac.lab/test/assets/test_rigid_object.py]: moved spawning to the function

File: source/extensions/omni.isaac.lab/test/assets/test_rigid_object.py [hunk 2]
```diff
@@ -79,6 +79,58 @@ def generate_cubes_scene(
...
+    cube_object_cfg_2 = RigidObjectCfg(
+        prim_path="/World/Table_1/Object",
+        spawn=spawn_cfg_2,
+        init_state=RigidObjectCfg.InitialStateCfg(pos=(0.0, 0.0, height)),
+    )
+    cube_object_cfg = RigidObjectCfg(
+        prim_path="/World/Table_.*/Object",
+        spawn=None,
+        init_state=RigidObjectCfg.InitialStateCfg(pos=(0.0, 0.0, height)),
+        use_first_matching_path=False,
```
Comment (Mayankm96) [inline source/extensions/omni.isaac.lab/test/assets/test_rigid_object.py:124]: I am a bit confused. Why is this needed? I thought this case is automatically handled. Otherwise `/World/envs/env_.*/Object` would have never worked.

File: source/extensions/omni.isaac.lab/test/assets/test_rigid_object.py [hunk 3]
```diff
@@ -79,6 +79,58 @@ def generate_cubes_scene(
...
+    cube_object_cfg_1 = RigidObjectCfg(
+        prim_path="/World/Table_0/Object",
+        spawn=spawn_cfg_1,
+        init_state=RigidObjectCfg.InitialStateCfg(pos=(0.0, 0.0, height)),
+    )
+    cube_object_cfg_2 = RigidObjectCfg(
+        prim_path="/World/Table_1/Object",
+        spawn=spawn_cfg_2,
+        init_state=RigidObjectCfg.InitialStateCfg(pos=(0.0, 0.0, height)),
+    )
```
Comment (Mayankm96) [inline source/extensions/omni.isaac.lab/test/assets/test_rigid_object.py:119]:
```suggestion
   spawn_cfg_1.func(
        prim_path="/World/Table_0/Object",
        spawn_cfg_1,
        translation=(0.0, 0.0, height),
    )
    spawn_cfg_2.func(
        prim_path="/World/Table_1/Object",
        spawn_cfg_2,
        translation=(0.0, 0.0, height),
    )
```

**Comment by Mayankm96** [comment]
Closing this MR as we have different mechanism is place now: #1288
